src/tattoo.ts

```diff
@@ -333,29 +386,55 @@ async function analyzeRepos() {
   return analyzer;
 }
 
-function openRepo(cloneOptions: nodegit.CloneOptions, ghRepo: GitHub.Repo) {
+
+
+async function openRepo(cloneOptions: nodegit.CloneOptions,
+  ghRepo: GitHub.Repo,
+  branchConfig: BranchConfig): Promise<ElementRepo> {
   const dir = path.join("repos", ghRepo.name);
-  let repoPromise: Promise<nodegit.Repository>;
+  let repo: nodegit.Repository;
   if (util.existsSync(dir)) {
     let updatedRepo: nodegit.Repository;
-    repoPromise = nodegit.Repository.open(dir).then((repo) => {
+    repo = await nodegit.Repository.open(dir).then((repo) => {
         updatedRepo = repo;
         return cloneRateLimiter.schedule(() =>
           updatedRepo.fetchAll(cloneOptions.fetchOpts)
         );
       }
     ).then(() => updatedRepo);
   } else {
-    repoPromise = cloneRateLimiter.schedule(() => {
+    repo = await cloneRateLimiter.schedule(() => {
       return nodegit.Clone.clone(
         ghRepo.clone_url,
         dir,
         cloneOptions);
     });
   }
-  return repoPromise.then((repo) =>
-    new ElementRepo({repo, dir, ghRepo, analyzer: null})
-  );
+  let repoConfig = branchConfig[ghRepo.name];
+  if (repoConfig && (repoConfig["branch"] || repoConfig["ref"])) {
+    const ref = repoConfig["branch"] || repoConfig["ref"];
```

Comment:
Seems like this part would be a little simpler if you divided the steps differently:

```typescript
  let ref = (repoConfig && (repoConfig["branch"] || repoConfig["ref"]))
      || getLatestRelease(repo)
      || 'master';
  repo = await checkout(repo, ref);
```